File: ui/diabetestracker-ui-2/src/app/models/dt-user-daily-data.ts
export class DTUserDailyData {
    date: Date;
    bloodSugarAM: number;
    breakfast: string;
    breakfastTime: string;
    lunch: string;
    lunchTime: string;
    dinner: string;
    dinnerTime: string;
    bloodSugarPM: number;

    constructor(
        date: Date,
        bloodSugarAM: number,
        breakfast: string,
        breakfastTime: string,
        lunch: string,
        lunchTime: string,
        dinner: string,
        dinnerTime: string,
        bloodSugarPM: number
      ) {
        this.date = date;
        this.bloodSugarAM = bloodSugarAM;
        this.breakfast = breakfast;
        this.breakfastTime = breakfastTime;
        this.lunch = lunch;
        this.lunchTime = lunchTime;
        this.dinner = dinner;
        this.dinnerTime = dinnerTime;
        this.bloodSugarPM = bloodSugarPM;
      }
  }
  

File: src/main/java/com/dharani/diabetestracker/entities/DTUserDailyData.java
package com.dharani.diabetestracker.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Table(name = "dt_user_daily_data")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DTUserDailyData {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    private String userName;
    private LocalDate date;
    private int bloodSugarAM;
    private String breakfast;
    private String breakfastTime;
    private String lunch;
    private String lunchTime;
    private String dinner;
    private String dinnerTime;
    private int bloodSugarPM;
}
